---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  fig.dev = "grDevices::png",
  dpi = 300L,
  dev.args = list(),
  fig.ext = "png",
  fig.height=8,
  fig.width=12,
  fig.retina = 2L,
  fig.align = "center"
)
```

# scstruc

The package for analysing the gene regulatory network based on the structure of single-cell transcriptomics data, inferred by Bayesian network. The function works with `SingleCellExperiment` and `SpatialExperiment`.

## Installation

Using `devtools`:

```{r, eval=FALSE}
devtools::install_github("noriakis/scstruc")
```

For using the CCDr algorithm in the latest R version, please install `ccdrAlgorithm` package from the following repository, which removes the `||` usage error.

```{r, eval=FALSE}
devtools::install_github("noriakis/ccdrAlgorithm")
```

## Examples

```{r message=FALSE, warning=FALSE, fig.width=8, fig.height=5}
library(scran)
library(scstruc)
library(bnlearn)
sce <- mockSCE()
sce <- logNormCounts(sce)
included_genes <- sample(row.names(sce), 100)
gs <- globalStruc(sce, included_genes, return_bn=TRUE, return_data=TRUE, change_symbol=FALSE, reg=FALSE)
fitted <- bn.fit(gs[[1]], gs[[2]])
ggraph(bn_fit_to_igraph(fitted), layout="fr") + 
  geom_edge_diagonal(aes(color=coef),
      arrow=arrow(type="closed", length=unit(2,"mm")),
      start_cap=circle(1,"mm"), end_cap=circle(1,"mm"))+
  geom_node_point()+
  theme_graph()
```

Using the bootstrapping, the averaged network is obtained.

```{r message=FALSE, warning=FALSE, fig.width=8, fig.height=8}
library(glmnet)
gs2 <- globalStruc(sce, included_genes, return_bn=TRUE, penalty="glmnet_BIC.boot",
                  change_symbol = FALSE, algorithm.args=list("R"=100))
plotAVN(gs2)
```

Celltype-specific bootstrapped networks.

```{r message=FALSE, warning=FALSE, fig.width=8, fig.height=8}
sce <- mockSCE()
sce2 <- mockSCE()
sces <- list(sce, sce2)
sces <- lapply(sces, function(x) logNormCounts(x))
sces <- lapply(sces, function(x) {
  colLabels(x) <- sample(c("Celltype_1","Celltype_2"), ncol(x), replace=TRUE)
  return(x)})

included_genes <- sample(row.names(sce), 50)
booted <- lapply(sces, function(x) perLabelStruc(x, label_name = "Celltype_1",
     algorithm.args=list("R"=10), barcode_column="row",
     candidate_genes=included_genes, nonzero=0.5,
     penalty="glmnet_CV.boot"))
ce <- coreEdges(booted, "Celltype_1")
ce
```
